tests/test_gaap.py

```diff
@@ -0,0 +1,183 @@
+            refCat = afwTable.SourceCatalog(refSchema)
+            refSource = refCat.addNew()
+            refSource.set(centroidKey, center + offset)
+            refSource.set(shapeKey, afwGeom.Quadrupole(1.0, 1.0, 0.0))
+
+            refSource.setCoord(refWcs.pixelToSky(refSource.get(centroidKey)))
+            taskInitArgs = (refSchema,)
+            taskRunArgs = (refCat, refWcs)
+        else:
+            taskInitArgs = (afwTable.SourceTable.makeMinimalSchema(),)
+            taskRunArgs = ()
+
+        # Activate undeblended measurement with the same configuration
+        measConfig.undeblended.names.add(algName)
+        measConfig.undeblended[algName] = measConfig.plugins[algName]
+
+        algMetadata = dafBase.PropertyList()
+        task = TaskClass(*taskInitArgs, config=measConfig, algMetadata=algMetadata)
+
+        schema = task.schema
+        measCat = afwTable.SourceCatalog(schema)
+        source = measCat.addNew()
+        source.getTable().setMetadata(algMetadata)
+        ss = afwDetection.FootprintSet(exposure.getMaskedImage(), afwDetection.Threshold(0.1))
+        fp = ss.getFootprints()[0]
+        source.setFootprint(fp)
+
+        task.run(measCat, exposure, *taskRunArgs)
+
+        disp = afwDisplay.Display(frame)
+        disp.mtv(exposure)
+        disp.dot("x", *center, origin=afwImage.PARENT, title="psfSigma=%f" % (psfSigma,))
+
+        self.assertFalse(source.get(algName + "_flag"))  # algorithm succeeded
+
+        # We will check the accuracy of the measured flux in a later ticket.
+        # We simply check now if it produces a positive number (non-nan)
+        for sF in algConfig.scalingFactors:
+            for sigma in algConfig.sigmas:
+                baseName = getGaapResultName(sF, sigma, algName)
+                self.assertTrue((source.get(baseName + "_instFlux") >= 0))
+                self.assertTrue((source.get(baseName + "_instFluxErr") >= 0))
+
+    def runGaap(self, forced, scalingFactors=(1.0, 1.1, 1.15, 1.2, 1.5, 2.0), psfSigmas=(1.7, 0.95, 1.3,)):
+        for psfSigma in psfSigmas:
+            self.check(psfSigma=psfSigma, forced=forced, scalingFactors=scalingFactors)
+
+    def testGaapUnforced(self):
+        self.runGaap(False)
+
+    def testGaapForced(self):
+        self.runGaap(True)
+
+
+class TestMemory(lsst.utils.tests.MemoryTestCase):
+    pass
+
+
+def setup_module(module, backend="virtualDevice"):
+    lsst.utils.tests.init()
+    try:
+        afwDisplay.setDefaultBackend(backend)
+    except Exception:
+        print("Unable to configure display backend: %s" % backend)
+
+
+if __name__ == "__main__":
+    import sys
+
+    from argparse import ArgumentParser
+    parser = ArgumentParser()
+    parser.add_argument('--backend', type=str, default="virtualDevice",
+                        help="The backend to use, e.g. 'ds9'. Be sure to 'setup display_<backend>'")
+    args = parser.parse_args()
```

Comment:
I confess I have no idea if this is the right way to set up display usage in a unit test, and if you weren't setting up a display, there are parts of it I might flag as potentially obsolete or unnecessary (but I'm not really sure about that either).  If this comes pretty much verbatim from some older file (and it's the same one you got those `__future__` imports from), we may want to look into whether it's still the right prescription.

Reply:
I have a display setup to see how (non-)Gaussian an image looks after PSF Gaussianization. The display can be dropped once the development is nearly complete, or at least after DM-27482 is done.